src/mlpack/methods/ann/loss_functions/hinge_embedding_loss_impl.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file hinge_embedding_loss_impl.hpp
+ * @author Lakshya Ojha
+ *
+ * Implementation of the Hinge Embedding loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_HINGE_EMBEDDING_LOSS_IMPL_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_HINGE_EMBEDDING_LOSS_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "hinge_embedding_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+HingeEmbeddingLoss<InputDataType, OutputDataType>::HingeEmbeddingLoss()
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double HingeEmbeddingLoss<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, const TargetType&& target)
```

Comment:
I think you missed out the recent change to method signatures in #2259 . You'll have to change your code to conform to it.

Reply:
Hi @favre49 
Thanks for noticing.
I have changed these in the new commit, kindly have a look when you get time.
Thanks.